server/region_handler.go

```diff
@@ -768,6 +874,125 @@ func (t *regionHandlerTool) getMvccByStartTs(startTS uint64, startKey, endKey []
+func (t *regionHandlerTool) formValue2DatumRow(values url.Values, idxCols []*model.ColumnInfo) ([]types.Datum, error) {
+	data := make([]types.Datum, len(idxCols))
+	for i, col := range idxCols {
+		colName := col.Name.String()
+		vals, ok := values[colName]
+		if !ok {
+			return nil, errors.BadRequestf("Missing value for index column %s.", colName)
+		}
+
+		switch len(vals) {
+		case 0:
+			data[i].SetNull()
+		case 1:
+			val := vals[0]
+			switch col.Tp {
+			case mysql.TypeTiny, mysql.TypeShort, mysql.TypeInt24, mysql.TypeLong, mysql.TypeLonglong, mysql.TypeYear:
+				if mysql.HasUnsignedFlag(col.Flag) {
+					elem, err := strconv.ParseUint(val, 10, 64)
+					if err != nil {
+						return nil, errors.Trace(err)
+					}
+					data[i].SetUint64(elem)
+				} else {
+					elem, err := strconv.ParseInt(val, 10, 64)
+					if err != nil {
+						return nil, errors.Trace(err)
+					}
+					data[i].SetInt64(elem)
+				}
+			case mysql.TypeFloat:
+				elem, err := strconv.ParseFloat(val, 32)
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+				data[i].SetFloat32(float32(elem))
+			case mysql.TypeDouble:
+				elem, err := strconv.ParseFloat(val, 64)
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+				data[i].SetFloat64(elem)
+			case mysql.TypeVarchar, mysql.TypeVarString, mysql.TypeString,
+				mysql.TypeBlob, mysql.TypeTinyBlob, mysql.TypeMediumBlob, mysql.TypeLongBlob:
+				data[i].SetBytes([]byte(val))
+			case mysql.TypeDate, mysql.TypeDatetime, mysql.TypeTimestamp:
+				elem, err := types.ParseTime(nil, val, col.Tp, col.Decimal)
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+				data[i].SetMysqlTime(elem)
+			case mysql.TypeDuration:
+				elem, err := types.ParseDuration(val, col.Decimal)
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+				data[i].SetMysqlDuration(elem)
+			case mysql.TypeNewDecimal:
+				var elem types.MyDecimal
+				err := elem.FromString([]byte(val))
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+				data[i].SetMysqlDecimal(&elem)
+			case mysql.TypeEnum:
+				num, err := strconv.ParseUint(val, 10, 64)
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+				elem, err := types.ParseEnumValue(col.Elems, num)
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+				data[i].SetMysqlEnum(elem)
+			case mysql.TypeSet:
+				num, err := strconv.ParseUint(val, 10, 64)
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+				elem, err := types.ParseSetValue(col.Elems, num)
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+				data[i].SetMysqlSet(elem)
+			case mysql.TypeBit:
+				data[i].SetMysqlBit([]byte(val))
+			case mysql.TypeJSON:
+				elem, err := tjson.ParseFromString(val)
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+				data[i].SetMysqlJSON(elem)
+			default:
+				return nil, errors.NotSupportedf("Not supported datum type %d.", col.Tp)
+			}
```

Comment:
Can we use the function of `ConvertTo` instead of line908-987?

Reply:
Done.